www/src/components/repository/RepositoryEdit.tsx

```diff
@@ -0,0 +1,677 @@
+  }
+}
+
+function RepoIcon({
+  src = null,
+  onClick,
+}: {
+  src: string | null;
+  onClick: EventHandler<any>;
+}) {
+  if (!src) {
+    return (
+      <Flex
+        width={96}
+        height={96}
+        backgroundColor="fill-two"
+        _hover={{
+          backgroundColor: 'fill-two-hover',
+        }}
+        border="1px solid border-fill-two"
+        borderRadius="medium"
+        align="center"
+        justify="center"
+        cursor="pointer"
+        onClick={onClick}
+      >
+        <P
+          title2
+          color="text-light"
+        >
+          +
+        </P>
+      </Flex>
+    )
+  }
+
+  return (
+    <Img
+      src={src}
+      alt="Icon"
+      width={64 + 32}
+      height={64 + 32}
+      objectFit="cover"
+      backgroundColor="fill-one"
+      cursor="pointer"
+      border="1px solid border"
+      onClick={onClick}
+    />
+  )
+}
+
+type FormState = {
+  name: string;
+  description: string;
+  category: string;
+  oauthUrl: string;
+  oauthMethod: string;
+  tags: { name: string }[];
+  private: boolean;
+};
+
+function RepositoryEdit() {
+  const {
+    id,
+    name,
+    description,
+    category,
+    editable,
+    oauthSettings,
+    tags,
+    icon,
+    private: privateRepo,
+  } = useContext(RepositoryContext) as any
+  const {
+    state: formState,
+    initialState: formInitialState,
+    hasUpdates: formStateHasUpdates,
+    update: updateFormState,
+    // reset: resetFormState,
+  } = useUpdateState<FormState>(
+    useMemo(
+      () => ({
+        name,
+        description,
+        category,
+        oauthUrl:
+          typeof oauthSettings?.uriFormat === 'string'
+            ? oauthSettings?.uriFormat
+            : '',
+        oauthMethod:
+          typeof oauthSettings?.authMethod === 'string'
+            ? oauthSettings?.authMethod
+            : authMethods.BASIC,
+        tags: isArray(tags) ? tags.map(tag => ({ name: tag.name })) : [],
+        private: !!privateRepo,
+      }),
+      [
+        name,
+        description,
+        oauthSettings?.authMethod,
+        oauthSettings?.uriFormat,
+        tags,
+        privateRepo,
+        category,
+      ]
+    )
+  )
+
+  const [tagSearchString, setTagSearchString] = useState('')
+  const newTag = tagSearchString
+    .replaceAll(/([\s_]+)/gu, '-')
+    .replaceAll(/[^A-Za-z-]+/gu, '')
+    .toLowerCase()
+  const tagSearch = useQuery(TAGS_SEARCH_QUERY, {
+    variables: { q: tagSearchString, first: 200 },
+  })
+  const tagSearchResults: { tag: string; count: number }[] =
+    tagSearch?.data?.tags?.edges?.map((edge: any) => edge?.node) || []
+
+  const [iconUpdate, setIconUpdate] = useState<{
+    file: File | null;
+    previewUrl: string | null;
+  }>({ file: null, previewUrl: icon || null })
+
+  const [, setSuccess] = useState(false)
+  const [, setError] = useState<ApolloError>()
+
+  const tagSearchRef = useRef<any>(null)
+
+  const [mutation, { loading }] = useMutation(UPDATE_REPOSITORY_MUTATION, {
```

Comment:
you can actually just get the error by doing `const [mutation, { loading, error }]` here